File: eis2/src/app/service/training-specialist-settings.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpHeaders, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs/Observable';
import { DialogueVo } from '../data/dialogue/dialoguevo';
import { TrainingSettingsData } from '../data/rest/training-settings-data';

@Injectable({
  providedIn: 'root'
})
export class TrainingSpecialistSettingsService {

  constructor(private http: HttpClient) { }

  getTrainingSettings(incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentId)
      .set('incidentGroupId', incidentGroupId);
    return this.http.get<DialogueVo>(`/service/training-specialists/settings`, {params: params});   
  }  

  saveTrainingSettings(data: TrainingSettingsData, incidentGroupId, incidentIds): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
      .set('incidentId', incidentIds)
      .set('incidentGroupId', incidentGroupId);
    return this.http.post<DialogueVo>(`/service/training-specialists/settings`, data, {params: params});
  }

  getTrainingContactGrids(incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
    .set('incidentId', incidentId)
    .set('incidentGroupId', incidentGroupId);
    return this.http.get<DialogueVo>(`/service/training-specialists/settings/training-contacts/grid`, {params: params});     
  }

  deleteTrainingContact(trainingContactId, incidentResourceId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
    .set('incidentResourceId', incidentResourceId);
    return this.http.delete<DialogueVo>(`/service/training-specialists/settings/training-contacts/${trainingContactId}`, {params: params});     
  }

  saveTrainingContact(data): Observable<DialogueVo> {
    return this.http.post<DialogueVo>(`/service/training-specialists/settings/training-contacts/`, data);     
  }  

  getPriorityProgramGrid(incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
    .set('incidentId', incidentId)
    .set('incidentGroupId', incidentGroupId);
    return this.http.get<DialogueVo>(`/service/training-specialists/settings/priority-programs/grid`, {params: params});     
  }  

  savePriorityProgram(data, incidentId, incidentGroupId): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
    .set('incidentId', incidentId)
    .set('incidentGroupId', incidentGroupId);
    return this.http.post<DialogueVo>(`/service/training-specialists/settings/priority-programs/`, data, {params: params});     
  }  

  deletePriorityProgram(priorityProgramId, incidentId, incidentGroupId, code): Observable<DialogueVo> {
    const params: HttpParams = new HttpParams()
    .set('incidentId', incidentId)
    .set('incidentGroupId', incidentGroupId)
    .set('code', (code));
    return this.http.delete<DialogueVo>(`/service/training-specialists/settings/priority-programs/${priorityProgramId}`, {params: params});     
  }

  saveAcres(data: TrainingSettingsData): Observable<DialogueVo> {
    return this.http.post<DialogueVo>(`/service/training-specialists/settings/acres/`, data);     
  }
}

File: isuite-core-ui/src/main/java/gov/nwcg/isuite/core/service/TrainingSpecialistSettingsService.java
package gov.nwcg.isuite.core.service;

import java.util.Collection;

import gov.nwcg.isuite.core.vo.PriorityProgramVo;
import gov.nwcg.isuite.core.vo.TrainingContactVo;
import gov.nwcg.isuite.core.vo.TrainingSettingsVo;
import gov.nwcg.isuite.core.vo.dialogue.DialogueVo;
import gov.nwcg.isuite.framework.core.service.TransactionService;
import gov.nwcg.isuite.framework.exceptions.ServiceException;

public interface TrainingSpecialistSettingsService extends TransactionService {
	
	public DialogueVo savePriorityProgram(PriorityProgramVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo deletePriorityProgram(PriorityProgramVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getPriorityProgramGrid(DialogueVo dialogueVo, Long incidentId, Long incidentGroupId) throws ServiceException;
	
	public DialogueVo getTrainingSettings(DialogueVo dialogueVo, Long incidentId, Long incidentGroupId) throws ServiceException;
	
//	public DialogueVo saveTrainingSettings(TrainingSettingsVo vo, DialogueVo dialogueVo) throws ServiceException;
	public DialogueVo saveTrainingSettings(TrainingSettingsVo vo, Long igId, Collection<Integer> incIds,  DialogueVo dialogueVo) throws ServiceException;
	
//	public DialogueVo getContactResourcesGrid(DialogueVo dialogueVo, Long incidentId, Long incidentGroupId) throws ServiceException;
	
	public DialogueVo saveTrainingContact(TrainingContactVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo deleteTrainingContact(TrainingContactVo vo, DialogueVo dialogueVo) throws ServiceException;
	
	public DialogueVo getTrainingContactGrids(DialogueVo dialogueVo, Long incidentId, Long incidentGroupId) throws ServiceException;
	
	public DialogueVo saveAcres(TrainingSettingsVo vo, DialogueVo dialogueVo) throws ServiceException;

}
